_posts/2024-09-12-opensearch-project-roadmap-2024-2025.md

```diff
@@ -0,0 +1,187 @@
+
+
+### Roadmap Theme 7: Security
+
+Security is a Tier 0 prerequisite for modern workloads. In OpenSearch, security features are primarily implemented by the Security plugin, which offers a rich set of capabilities. These include various authentication backends (SAML, JWT, LDAP), authorization primitives, fine-grained access control (DLS/FLS), and encryption in transit. OpenSearch has [rapidly developed new plugin capabilities](https://github.com/orgs/opensearch-project/projects/206/views/11?sliceBy%5Bvalue%5D=Security), attracting increased interest from the community. This growth also raises critical security implications. Importantly, security should not come at the cost of performance. To address these challenges, OpenSearch is focusing on the following initiatives to strengthen its security posture.
+
+[**Plugin resource permissions**](https://github.com/opensearch-project/security/issues/4500): We are developing a mechanism for sharing plugin resources that supports existing use cases while allowing more granular control over resource sharing. Examples include model groups in the ML Commons plugin, anomaly detectors in the Time Series Analytics plugin, and detectors in the Alerting plugin.
+
+[**Plugin isolation**](https://github.com/opensearch-project/security/issues/4439): OpenSearch is moving towards a zero-trust model for plugins. Cluster administrators will have full [visibility into all permissions](https://github.com/opensearch-project/security/issues/2860) requested by a plugin before installation.
+
+[**Optimized privilege evaluation**](https://github.com/opensearch-project/security/issues/3870): Performance is a key focus for OpenSearch. We’ve identified areas within the Security plugin that can yield significant performance improvements, especially for clusters with numerous indexes or roles mapped to users.
+
+[**API tokens**](https://github.com/opensearch-project/security/issues/4009): API tokens introduce a new way to interact with OpenSearch clusters by associating permissions directly with a token. Cluster administrators will have full visibility and control over the issued tokens and their usage.
+
+[**Ease of use**](https://github.com/opensearch-project/security-dashboards-plugin/issues/2070): We aim to simplify security setup for cluster administrators. Many useful security features remain underused because they are not exposed through OpenSearch Dashboards. To address this, we will add security dashboard pages, where administrators can configure rate limiters to protect clusters from unauthenticated actors.
```

Comment:
```suggestion
[**Ease of use**](https://github.com/opensearch-project/security-dashboards-plugin/issues/2070): We aim to simplify security setup for cluster administrators. Many useful security features remain underused because they are not exposed through OpenSearch Dashboards. To address this, we will add security dashboard pages where administrators can configure rate limiters to protect clusters from unauthenticated actors.
```